src/core/pointcloud/qgspointclouddataprovider.cpp

```diff
@@ -176,3 +181,94 @@ QVariant QgsPointCloudDataProvider::metadataClassStatistic( const QString &, con
 {
   return QVariant();
 }
+
+QVector<QMap<QString, QVariant>> QgsPointCloudDataProvider::identify(
+                                QgsPointCloudLayer *layer,
+                                float maximumError,
+                                float rootErrorPixels,
+                                QgsGeometry extentGeometry,
+                                const QgsDoubleRange extentZRange )
+{
+  QVector<QMap<QString, QVariant>> acceptedPoints;
+
+  QgsPointCloudIndex *index = this->index();
+  const IndexedPointCloudNode root = index->root();
+
+
+  QgsPointCloudLayerElevationProperties *properties = dynamic_cast<QgsPointCloudLayerElevationProperties *>( layer->elevationProperties() );
+  QgsRenderContext renderContext;
+
+  QVector<IndexedPointCloudNode> nodes = traverseTree( index, root, maximumError, rootErrorPixels, extentGeometry, extentZRange );
+
+  QgsPointCloudAttributeCollection attributeCollection = index->attributes();
+  QgsPointCloudRequest request;
+  request.setAttributes( attributeCollection );
+
+  QgsPointCloudRenderContext context( renderContext, index->scale(), index->offset(), properties->zScale(), properties->zOffset() );
```

Comment:
Can you remove this? It's only used during the calculation of the point x/y, and that can be directly calculated using the index's offset/scaling. QgsPointCloudRenderContext is a renderer specific class, and we shouldn't be using it outside of rendering.